sklearn/pipeline.py

```diff
@@ -226,14 +243,12 @@ def _fit(self, X, y=None, **fit_params):
             step, param = pname.split('__', 1)
             fit_params_steps[step][param] = pval
         Xt = X
-        for name, transform in self.steps[:-1]:
+        for idx_tr, (name, transform) in enumerate(self.steps[:-1]):
```

Comment:
I would rather rename `idx_tr` to `step_idx` (or `transformer_idx` but I tend to prefer `step_idx`).